### PR TITLE
Add APIs for campaigns

### PR DESCRIPTION
**Pull request by minsikzzang**
Add APIs for campaigns

Related Doc
- http://naver.github.io/searchad-apidoc/#/tags/Campaign

Tasks
- [x] Complete Authorization for Naver API
- [x] Add Naver Campaign API
- [x] Refactor existing unit tests

### CODE REVIEW

File: lib/naver/searchad/api/auth.rb [hunk 1]
```diff
@@ -8,19 +8,37 @@ class CustomerAcccountCredentials
...
 
-          def generate_signature
+          def generate_signature(secret, request_uri, method, timstamp)
+            method = method.to_s.upcase if method.is_a?(Symbol)
+
+            Base64.encode64(
+              OpenSSL::HMAC.digest(
+                OpenSSL::Digest::SHA256.new,
+                secret,
+                "#{timstamp}.#{method}.#{request_uri}"
```
Comment (robertino) [inline lib/naver/searchad/api/auth.rb]: [timestamp, method, request_uri].join('.') ?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -11,8 +11,14 @@
...
-require "naver/searchad/api"
-require "naver/searchad/api/core/base_service"
+require 'naver/searchad/api'
+require 'naver/searchad/api/core/base_service'
+require 'naver/searchad/api/auth'
+require 'naver/searchad/api/campaign/service'
+
+ENV['NAVER_API_KEY'] = '0100000000f2e75122770874cb904034e7e27f5815c21af53a93f25b0f05f0ce97f263650c'
+ENV['NAVER_API_SECRET'] = 'AQAAAADy51Eidwh0y5BANOfif1gVwHjWG4MrXg6Mbh54YHY4MQ=='
+ENV['NAVER_API_CLIENT_ID'] = '1077530'
```
Comment (robertino) [inline spec/spec_helper.rb]: what are these keys above?
are they supposed to be in a public repo?

Comment (minsikzzang) [inline spec/spec_helper.rb]: this should be testing credentials, I just put them to run to collect data for mockup, should be removed later

File: lib/naver/searchad/api/auth.rb (hunk 1 above)
Comment (chiting) [inline lib/naver/searchad/api/auth.rb]: Hmm, did a quick bench and https://gist.github.com/chiting/d55972931a7fc2904fecf5c66463983f

File: lib/naver/searchad/api/auth.rb [hunk 3]
```diff
@@ -8,19 +8,35 @@ class CustomerAcccountCredentials
...
           private
 
-          def generate_signature
+          def generate_signature(secret, request_uri, method, timestamp)
+            method = method.to_s.upcase if method.is_a?(Symbol)
+
+            Base64.encode64(OpenSSL::HMAC.digest(
+              OpenSSL::Digest::SHA256.new,
+              secret,
+              [timestamp, method, request_uri].join('.')
```
Comment (chiting) [inline lib/naver/searchad/api/auth.rb]: Hmm, did a quick bench (ruby 2.4.1) and https://gist.github.com/chiting/d55972931a7fc2904fecf5c66463983f, no strong opinion as both are as readable to me but at least you know :)

Comment (minsikzzang) [inline lib/naver/searchad/api/auth.rb]: I do not have strong opinion either, kinda like both styles `[a, b, c].join(',')` and `"#{a}.#{b}.#{c}"` @robertino what do you think?

Comment (minsikzzang) [inline lib/naver/searchad/api/auth.rb]: should we choose then one with faster way? @chiting @robertino

Comment (minsikzzang) [inline lib/naver/searchad/api/auth.rb]: https://gist.github.com/norman/1192140 it seems early version of ruby, there are no difference, latest version of ruby, some performance difference between two ways

Comment (robertino) [inline lib/naver/searchad/api/auth.rb]: I think the join looks cleaner, speed doesn't really make much differenece here as it's generating signature for http call

Comment (chiting) [inline lib/naver/searchad/api/auth.rb]: ok then, `join` it is :)

**Pull request merged**